examples/instructlab-qnagen.ipynb

```diff
@@ -0,0 +1,283 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "78d52cfe-70b3-4d93-8fc2-f28366e2bde2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install -qq -e .."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1b7d8e61-379e-424a-a66c-58224fe099d3",
+   "metadata": {},
+   "source": [
+    "# Find PDF files from data/ subdirectory and convert using Docling"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "1f012ac3-eb4d-4e22-9281-520ea2e573d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import glob\n",
+    "\n",
+    "from docling.document_converter import DocumentConverter\n",
+    "\n",
+    "input_files = glob.glob(f\"data/*.pdf\")\n",
+    "\n",
+    "doc_converter = DocumentConverter()\n",
+    "docs = doc_converter.convert_all(input_files)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3bbe7773-2c0f-4397-8df8-986a12b939b6",
+   "metadata": {},
+   "source": [
+    "# Chunk each document and apply filters to chunks"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "ed7115aa-2782-4482-a0af-9b8c03c7a7b3",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Chunking and filtering document oa_pdf_00_c3_gkl753_PMC1781121\n",
+      "Created dataset oa_pdf_00_c3_gkl753_PMC1781121 with 16 QA chunks\n",
+      "Chunking and filtering document oa_pdf_00_c5_main_PMC467086\n",
+      "Created dataset oa_pdf_00_c5_main_PMC467086 with 9 QA chunks\n"
+     ]
+    }
+   ],
+   "source": [
+    "from docling_core.transforms.chunker.hierarchical_chunker import HierarchicalChunker\n",
+    "from docling_sdg.qa.utils import get_qa_chunks\n",
+    "\n",
+    "chunker = HierarchicalChunker()\n",
```

Comment:
I feel like you could probably get more usable chunks here if you switched to the hybrid chunker, because the hybrid chunker will merge hierarchical chunks (at least within a section) when there are multiple in a row that all fit within the target length.   So you're more likely to see more chunks longer than 500 characters if you use the hybrid chunker and set the target length to something reasonable.